From be3355235a57640afb455cf02d2ce183040796e1 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 5 Feb 2025 09:22:25 +0000
Subject: [PATCH] arm64: dts: Clean up the downstream patches

Remove some gratuitous differences with the upstream dts, and drop the
unnecessary 'status = "okay"' properties (they are needed(*) to override
'status = "disabled"' in the original definitions.

(*) You could technically delete the original properties, but that looks
worse and doesn't work in an overlay.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 .../boot/dts/broadcom/bcm2712-rpi-5-b.dts     | 33 ++++++++-----------
 .../boot/dts/broadcom/bcm2712-rpi-cm5.dtsi    | 25 +++++++-------
 arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi |  6 ----
 3 files changed, 25 insertions(+), 39 deletions(-)

--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: GPL-2.0
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
 /dts-v1/;
 
 #include <dt-bindings/gpio/gpio.h>
@@ -42,7 +42,7 @@
 		};
 	};
 
-	sd_io_1v8_reg: sd_io_1v8_reg {
+	sd_io_1v8_reg: sd-io-1v8-reg {
 		compatible = "regulator-gpio";
 		regulator-name = "vdd-sd-io";
 		regulator-min-microvolt = <1800000>;
@@ -51,12 +51,11 @@
 		regulator-always-on;
 		regulator-settling-time-us = <5000>;
 		gpios = <&gio_aon 3 GPIO_ACTIVE_HIGH>;
-		states = <1800000 0x1
-			  3300000 0x0>;
-		status = "okay";
+		states = <1800000 1>,
+			 <3300000 0>;
 	};
 
-	sd_vcc_reg: sd_vcc_reg {
+	sd_vcc_reg: sd-vcc-reg {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc-sd";
 		regulator-min-microvolt = <3300000>;
@@ -64,10 +63,9 @@
 		regulator-boot-on;
 		enable-active-high;
 		gpios = <&gio_aon 4 GPIO_ACTIVE_HIGH>;
-		status = "okay";
 	};
 
-	wl_on_reg: wl_on_reg {
+	wl_on_reg: wl-on-reg {
 		compatible = "regulator-fixed";
 		regulator-name = "wl-on-regulator";
 		regulator-min-microvolt = <3300000>;
@@ -97,7 +95,6 @@
 		compatible = "regulator-fixed";
 		regulator-name = "cam0_reg";
 		enable-active-high;
-		status = "okay";
 		gpio = <&rp1_gpio 34 0>;  // CD0_IO0_MICCLK, to MIPI 0 connector
 	};
 
@@ -105,14 +102,12 @@
 		compatible = "regulator-fixed";
 		regulator-name = "cam1_reg";
 		enable-active-high;
-		status = "okay";
 		gpio = <&rp1_gpio 46 0>;  // CD1_IO0_MICCLK, to MIPI 1 connector
 	};
 
 	cam_dummy_reg: cam_dummy_reg {
 		compatible = "regulator-fixed";
 		regulator-name = "cam-dummy-reg";
-		status = "okay";
 	};
 
 	dummy: dummy {
@@ -131,11 +126,6 @@ rp1_target: &pcie2 {
 	status = "okay";
 };
 
-// Add some labels to 2712 device
-
-// The system UART
-&uart10 { status = "okay"; };
-
 // The system SPI for the bootloader EEPROM
 &spi10 { status = "okay"; };
 
@@ -188,6 +178,13 @@ rp1_target: &pcie2 {
 	};
 };
 
+/* The Debug UART, on Rpi5 it's on JST-SH 1.0mm 3-pin connector
+ * labeled "UART", i.e. the interface with the system console.
+ */
+&uart10 {
+	status = "okay";
+};
+
 gpio: &rp1_gpio {
 	status = "okay";
 };
@@ -716,10 +713,6 @@ spi10_cs_pins: &spi10_cs_gpio1 {};
 	clock-names = "hdmi", "bvb", "audio", "cec";
 };
 
-&pcie1 {
-	status = "okay";
-};
-
 &pcie2 {
 	status = "okay";
 };
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: GPL-2.0
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
 
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/clock/rp1.h>
@@ -41,7 +41,7 @@
 		};
 	};
 
-	sd_io_1v8_reg: sd_io_1v8_reg {
+	sd_io_1v8_reg: sd-io-1v8-reg {
 		compatible = "regulator-fixed";
 		regulator-name = "vdd-sd-io";
 		regulator-min-microvolt = <1800000>;
@@ -49,7 +49,7 @@
 		regulator-always-on;
 	};
 
-	sd_vcc_reg: sd_vcc_reg {
+	sd_vcc_reg: sd-vcc-reg {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc-sd";
 		regulator-min-microvolt = <3300000>;
@@ -57,10 +57,9 @@
 		regulator-boot-on;
 		enable-active-high;
 		gpios = <&gio_aon 4 GPIO_ACTIVE_HIGH>;
-		status = "okay";
 	};
 
-	wl_on_reg: wl_on_reg {
+	wl_on_reg: wl-on-reg {
 		compatible = "regulator-fixed";
 		regulator-name = "wl-on-regulator";
 		regulator-min-microvolt = <3300000>;
@@ -90,14 +89,12 @@
 		compatible = "regulator-fixed";
 		regulator-name = "cam0_reg";
 		enable-active-high;
-		status = "okay";
-		gpio = <&rp1_gpio 34 0>; // CD0_IO0_MICCLK, to CAM_GPIO on connector
+		gpio = <&rp1_gpio 34 0>;  // CD0_IO0_MICCLK, to CAM_GPIO on connector
 	};
 
 	cam_dummy_reg: cam_dummy_reg {
 		compatible = "regulator-fixed";
 		regulator-name = "cam-dummy-reg";
-		status = "okay";
 	};
 
 	dummy: dummy {
@@ -116,11 +113,6 @@ rp1_target: &pcie2 {
 	status = "okay";
 };
 
-// Add some labels to 2712 device
-
-// The system UART
-&uart10 { status = "okay"; };
-
 // The system SPI for the bootloader EEPROM
 &spi10 { status = "okay"; };
 
@@ -175,6 +167,13 @@ rp1_target: &pcie2 {
 	};
 };
 
+/* The Debug UART, on Rpi5 it's on JST-SH 1.0mm 3-pin connector
+ * labeled "UART", i.e. the interface with the system console.
+ */
+&uart10 {
+	status = "okay";
+};
+
 gpio: &rp1_gpio {
 	status = "okay";
 };
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
@@ -35,13 +35,11 @@
 	fb: fb {
 		compatible = "brcm,bcm2708-fb";
 		firmware = <&firmware>;
-		status = "okay";
 	};
 
 	rpi_rtc: rpi_rtc {
 		compatible = "raspberrypi,rpi-rtc";
 		firmware = <&firmware>;
-		status = "okay";
 		trickle-charge-microvolt = <0>;
 	};
 
@@ -54,28 +52,24 @@
 			compatible = "raspberrypi,rpi-otp";
 			firmware = <&firmware>;
 			reg = <0 192>;
-			status = "okay";
 		};
 
 		nvmem_cust: nvmem_cust {
 			compatible = "raspberrypi,rpi-otp";
 			firmware = <&firmware>;
 			reg = <1 8>;
-			status = "okay";
 		};
 
 		nvmem_mac: nvmem_mac {
 			compatible = "raspberrypi,rpi-otp";
 			firmware = <&firmware>;
 			reg = <2 6>;
-			status = "okay";
 		};
 
 		nvmem_priv: nvmem_priv {
 			compatible = "raspberrypi,rpi-otp";
 			firmware = <&firmware>;
 			reg = <3 16>;
-			status = "okay";
 		};
 	};
 
